### ISSUE TITLE
Bun install crashes when using production flag and workspaces

### ISSUE THREAD

**Issue by sebastianbuechler** [oven-sh/bun#11150]
### How can we reproduce the crash?

I have a setup with a workspace `package.json` in the root folder and some workspaces in the folders `apps` (for different apps as entrypoints) and `packages` (for reusable packages). 

When I run `bun install` or `bun install --frozen-lockfile` everything works fine, but `bun install --frozen-lockfile --production` or even just `bun install --production` bun crashes with this bun.report.

### Relevant log output

```shell
bun install --production
[0.00ms] ".env"
bun install v1.1.7 (b0b7db5c)

============================================================
Bun v1.1.7 (b0b7db5c) Windows x64
Args: "C:\Users\myuser\.bun\bin\bun.exe", "install", "--production"
Features: bunfig dotenv 
Elapsed: 138ms | User: 0ms | Sys: 78ms
RSS: 79.84MB | Peak: 79.84MB | Commit: 125.96MB | Faults: 19685

panic(main thread): Internal assertion failure
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.7/wi1b0b7db5AgKgin5L+7yuIwkz1F+oiC+zk1D0pnkEo2+iE8pvvFq4kBiiJA0eNrzzCtJLcpLzFFILC5OLSrJzM9TSEvMzCktSgUAiSkKPg
```

### Stack Trace (bun.report)

Bun v1.1.7 ([`b0b7db5`](https://github.com/oven-sh/bun/tree/b0b7db5c060e830801b0273b7ff6e34e3eb5c254)) on windows x86_64 [InstallCommand]

**panic**: Internal assertion failure

- [`debug.zig:373`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/deps/zig/lib/std/debug.zig#L373): `panicExtra`
- [`debug.zig:348`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/deps/zig/lib/std/debug.zig#L348): `panic`
- [`output.zig:356`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/output.zig#L356): `panic`
- [`bun.zig:3159`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/bun.zig#L3159): `assertionFailure`
- [`fmt.zig:653`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/deps/zig/lib/std/fmt.zig#L653): `formatType`
- [`install.zig:10960`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/install/install.zig#L10960): `installWithManager`
- [`filter_run.zig:428`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/cli/filter_run.zig#L428): `install`
- [`cli.zig:1440`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/cli.zig#L1440): `start`
- [`cli.zig:62`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/cli.zig#L62): `start`
- [`main.zig:51`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/main.zig#L51): `main`

**Comment by sebastianbuechler** [comment]
New try but same issue with bun v1.1.8:

Bun v1.1.8 ([`89d2580`](https://github.com/oven-sh/bun/tree/89d25807fbf4488d3bfcb116484d0c2cf824f98c)) on windows x86_64 [InstallCommand]

**panic**: Internal assertion failure

- [`debug.zig:373`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/deps/zig/lib/std/debug.zig#L373): `panicExtra`
- [`debug.zig:348`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/deps/zig/lib/std/debug.zig#L348): `panic`
- [`output.zig:356`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/output.zig#L356): `panic`
- [`bun.zig:3177`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/bun.zig#L3177): `assertionFailure`
- [`fmt.zig:653`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/deps/zig/lib/std/fmt.zig#L653): `formatType`
- [`install.zig:11022`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/install/install.zig#L11022): `installWithManager`
- [`filter_run.zig:428`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/cli/filter_run.zig#L428): `install`
- [`cli.zig:1440`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/cli.zig#L1440): `start`
- [`cli.zig:62`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/cli.zig#L62): `start`
- [`main.zig:51`](https://github.com/oven-sh/bun/blob/89d25807fbf4488d3bfcb116484d0c2cf824f98c/src/main.zig#L51): `main`

**Comment by sebastianbuechler** [comment]
Fixed in #11245 Thanks!
